packages/block-library/src/navigation-menu/edit.js

```diff
@@ -79,6 +95,7 @@ function NavigationMenu( {
 		'wp-block-navigation-menu', {
 			'has-text-color': textColor.color,
 			'has-background-color': backgroundColor.color,
+			'has-scroll-x': hasScrollX,
```

Comment:
This classname doesn't seem to be used for anything. If we can scrap it we can also get rid of all the state and effect logic above.